### PR TITLE
DDragon migration

### PR DESCRIPTION
**Pull request by JanOuborny**
Migrated the now deprecated static data endpoint to use the ddragon endpoint:

- Shortened endpoint methods from `GetChampionsAsync` to `GetAllAsync`
- Renamed static data endpoint property from `Static` to `StaticData` to match class naming and property naming
- All get by id method got removed

### CODE REVIEW

**Comment by JanOuborny** [comment]
@BenFradet When this PR is done and when we have resolved all the recent issues, we should publish a 3.1.0 version.

**Comment by BenFradet** [comment]
:+1: I need to find out if I can build the nuget from linux as I don't have a Windows partition anymore

**Comment by JanOuborny** [comment]
@BenFradet If you can't build it, I can build it for you and upload it to GitHub.

**Comment by BenFradet** [comment]
cool, I'll try it out and let you know :+1:

File: RiotSharp/Endpoints/StaticDataEndpoint/Realm/StaticRealmEndpoint.cs [hunk 1]
```diff
@@ -29,7 +28,7 @@ public StaticRealmEndpoint(IRateLimitedRequester requester, ICache cache)
                 return wrapper.RealmStatic;
             }
 
-            var json = await requester.CreateGetRequestAsync(StaticDataRootUrl + RealmsUrl, region).ConfigureAwait(false);
+            var json = await requester.CreateGetRequestAsync($"https://ddragon.leagueoflegends.com/realms/{region.ToString().ToLower()}.json").ConfigureAwait(false);
```
Comment (BenFradet) [inline RiotSharp/Endpoints/StaticDataEndpoint/Realm/StaticRealmEndpoint.cs]: could we extract this variable too?

File: RiotSharp/Endpoints/StaticDataEndpoint/StaticDataEndpoints.cs [hunk 2]
```diff
@@ -11,41 +11,36 @@ public class StaticDataEndpoints : IStaticDataEndpoints
...
         }
 
-        private StaticDataEndpoints(string apiKey, bool useCache = true)
+        private StaticDataEndpoints(bool useCache = true)
         {
-            Requesters.StaticApiRequester = new RateLimitedRequester(apiKey, new Dictionary<TimeSpan, int>
-            {
-                { new TimeSpan(1, 0, 0), 10 }
-            });
+            Requesters.StaticApiRequester = new Requester("NOT-NEEDED");
```
Comment (BenFradet) [inline RiotSharp/Endpoints/StaticDataEndpoint/StaticDataEndpoints.cs]: don't we have something not requiring an api key?

File: RiotSharp/Endpoints/StaticDataEndpoint/TarballLinks/StaticTarballLinkEndPoint.cs [hunk 3]
```diff
@@ -9,33 +9,11 @@
...
-            }
-
-            var json = await requester.CreateGetRequestAsync(StaticDataRootUrl + TarballLinksUrl, region,
-                    new List<string> { !string.IsNullOrEmpty(version) ? $"version={version}" : null }).ConfigureAwait(false);
-            var tarballLink = JsonConvert.DeserializeObject<string>(json);
-
-            cache.Add(cacheKey, tarballLink, SlidingExpirationTime);
-
-            return tarballLink;
+            return $"http://ddragon.leagueoflegends.com/cdn/dragontail-{version}.tgz";
```
Comment (BenFradet) [inline RiotSharp/Endpoints/StaticDataEndpoint/TarballLinks/StaticTarballLinkEndPoint.cs]: same here

File: RiotSharp/Endpoints/StaticDataEndpoint/Version/StaticVersionsEndpoint.cs [hunk 4]
```diff
@@ -11,26 +11,25 @@ namespace RiotSharp.Endpoints.StaticDataEndpoint.Version
...
+            var cacheKey = VersionsCacheKey;
             var wrapper = cache.Get<string, List<string>>(cacheKey);
             if (wrapper != null)
             {
                 return wrapper;
             }
 
             var json =
-                await requester.CreateGetRequestAsync(StaticDataRootUrl + VersionsUrl, region).ConfigureAwait(false);
+                await requester.CreateGetRequestAsync("https://ddragon.leagueoflegends.com/api/versions.json").ConfigureAwait(false);
```
Comment (BenFradet) [inline RiotSharp/Endpoints/StaticDataEndpoint/Version/StaticVersionsEndpoint.cs]: same

**Pull request merged**